### PR TITLE
DBZ-7107 Introduce micro/nano second based envelope timestamps

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-7107

- [ ] Update documentation
- [x] Create sibling PRs for Informix, Db2, Vitess, Spanner, and Cassandra.

@jpechane once we agree on the implementation, I'll address the documentation.

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/schema/SchemaFactory.java [hunk 1]
```diff
@@ -365,6 +365,8 @@ public Envelope.Builder withDoc(String doc) {
             public Envelope build() {
                 builder.field(Envelope.FieldName.OPERATION, Envelope.OPERATION_REQUIRED ? Schema.STRING_SCHEMA : Schema.OPTIONAL_STRING_SCHEMA);
                 builder.field(Envelope.FieldName.TIMESTAMP, Schema.OPTIONAL_INT64_SCHEMA);
+                builder.field(Envelope.FieldName.TIMESTAMP_US, Schema.OPTIONAL_INT64_SCHEMA);
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/schema/SchemaFactory.java:368]: Should we update the envelope schema version?

**Review by jpechane — COMMENTED** [review]
@Naros LGTM, thanks. But I wonder if we should make it symmetric with `source` info block too. Some of the sources also provide better precisision then just milliseconds.

**Comment by Naros** [comment]
All good points, I'll amend. :+1:

**Comment by Naros** [comment]
The test failures related to PostgreSQL, SQL Server, and Debezium Server are unrelated.  The failures related to Db2, Spanner, Informix, and Vitess have been fixed in separate pull requests.

**Comment by jpechane** [comment]
@Naros Applied, thanks!

**Comment by jpechane** [comment]
@Naros Please provide docs update as a separate PR

**Pull request merged**